app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -450,12 +450,16 @@ def rhsm_url
         end
       end
 
+      def api_url
+        URI("https://#{URI.parse(url).host}:8443")
```

Comment:
This isn't quite correct. It only works on content proxies. On a Foreman + Smart Proxy this doesn't exist. So for the common case where there's only a single server you're breaking things. There is no way to detect if this is present or not. It can even be changed via an installer argument.

Also note that in the future we want to move this service away. Then Foreman Proxy can run on port 8443, just as a vanilla Foreman.

This is precisely why we introduced the registration module in Smart Proxy: it's a proper service we can rely on.

Also why I'm hammering on that Foreman -> Client -> Foreman is such a bad idea.

Reply:
Hmm, so how to solve the problem with calling Foreman API here: https://github.com/Katello/katello/pull/9756/files#diff-dc69411f7cfa18aebde7d457a2d44098317c9d33755139b130ca5c8bd9fd88eaR42

Should we call only Foreman API with the risk that it is not accessible from the host?